### DISCUSSION TITLE
code search pagination don't work correctly.

### DISCUSSION THREAD

**Discussion by codestar73** [community/community#90722] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Code search only works for 5 pages. It disables the next button even though there is many result.
![image](https://github.com/community/community/assets/75422800/fbf2cbda-b519-4275-93d0-5d9bb5c902d7)

**Comment by georgetoloraia** [comment]
### The limitation you're experiencing with GitHub's code search functionality is a part of GitHub's efforts to manage the performance and the resources needed to provide search results. Here's a breakdown of the situation and some potential ways to work within these constraints:

1. Search Result Limitation:

- GitHub limits the number of search results you can view to the first 100 items (or the first 5 pages, with 20 results per page). This is likely designed to conserve server resources and ensure that the search function remains fast and reliable for all users.

2. Why This Happens:

- Serving large numbers of search results can be resource-intensive. By limiting the number of pages you can view, GitHub ensures that its search infrastructure remains efficient and responsive.
This limitation also helps in reducing the potential misuse of the search API (e.g., scraping, data mining) which can negatively impact the service for others.


3. Working Within the Limitation:

- Refine Your Search: Use more specific search queries. The more specific your search terms are, the more likely you are to find what you need within the first 100 results.

4. Use Advanced Search Queries:

- GitHub supports a range of qualifiers that you can use to narrow down your search. For example, you can search within a specific repository, user's repositories, or language.

5. Paginate Manually:

- If you're developing a tool or script and need to retrieve more than 100 results, consider using the GitHub API, which may offer more flexibility in terms of pagination and rate limits. Note, however, that the API also has rate limits and usage policies that you'll need to adhere to.

**Reply by sgf** [reply]
I really can't understand Microsoft's behavior. You don't reduce costs through technology, but rely on castration of functions and degrading the experience. This is really puzzling.

**Reply by kaiser101** [reply]
Typical Microsoft. The earlier philosophy was release a buggy product then iron out the bugs. Of late, its changed to release a buggy product then make it more buggier.
